js/NetworkPane.js

```diff
@@ -0,0 +1,221 @@
+  _paneRef = null;
+  _networkRef = null;
+  _width = null;
+  _height = null;
+
+  handleDownload = () => {
+    var link = document.createElement('a');
+    link.download = `${this.props.title || 'visdom_image'}.jpg`;
+    link.href = this.props.content.src;
+    link.click();
+  };
+
+  CreateNetwork = graph => {
+    var width = 500,
+      height = 500;
+    var color = d3.scale.category10();
+    var force = d3.layout
+      .force()
+      .charge(-120)
+      .linkDistance(120)
+      .size([width, height]);
+    var svg = d3.select('.Network_Div').select('svg');
+    if (svg.empty()) {
+      svg = d3
+        .select('.Network_Div')
+        .append('svg')
+        .attr('width', width)
+        .attr('height', height);
+    }
+
+    if (this.props.Gtype == 'dgph' || this.props.Gtype == 'dwgph') {
+      svg
+        .append('defs')
+        .append('marker')
+        .attrs({
+          id: 'arrowhead',
+          viewBox: '-0 -5 10 10',
+          refX: 13,
+          refY: 0,
+          orient: 'auto',
+          markerWidth: 13,
+          markerHeight: 13,
+          xoverflow: 'visible',
+        })
+        .append('svg:path')
+        .attr('d', 'M 0,-5 L 10 ,0 L 0,5')
+        .attr('fill', '#999')
+        .style('stroke', 'none');
+    }
+
+    force
+      .nodes(graph.nodes)
+      .links(graph.links)
+      .start();
+
+    var link = svg
+      .selectAll('.link')
+      .data(graph.links)
+      .enter()
+      .append('line')
+      .attr('class', 'link')
+      .attr('marker-end', 'url(#arrowhead)');
+
+    link.append('title').text(function(d) {
+      return d.type;
+    });
+
+    var edgepaths = svg
+      .selectAll('.edgepath')
+      .data(graph.links)
+      .enter()
+      .append('path')
+      .attrs({
+        class: 'edgepath',
+        'fill-opacity': 0,
+        'stroke-opacity': 0,
+        id: function(d, i) {
+          return 'edgepath' + i;
+        },
+      })
+      .style('pointer-events', 'none');
+
+    var edgelabels = svg
+      .selectAll('.edgelabel')
+      .data(graph.links)
+      .enter()
+      .append('text')
+      .style('pointer-events', 'none')
+      .attrs({
+        class: 'edgelabel',
+        id: function(d, i) {
+          return 'edgelabel' + i;
+        },
+        'font-size': 10,
+        fill: '#aaa',
+      });
+
+    if (this.props.showEdgeLabels) {
+      if (this.props.labels === 'custom') {
+        edgelabels
+          .append('textPath')
+          .attr('xlink:href', (d, i) => '#edgepath' + i)
+          .style('text-anchor', 'middle')
+          .style('pointer-events', 'none')
+          .attr('startOffset', '50%')
+          .text(d => d.label);
+      } else {
+        edgelabels
+          .append('textPath')
+          .attr('xlink:href', (d, i) => '#edgepath' + i)
+          .style('text-anchor', 'middle')
+          .style('pointer-events', 'none')
+          .attr('startOffset', '50%')
+          .text(d => d.label);
```

Comment:
the branches code is identical: the `if` does not do anything

Reply:
Thanks for pointing this out! I will make the changes.